packages/offix-client/src/OfflineClient.ts

```diff
@@ -135,14 +175,22 @@ export class OfflineClient implements ListenerProvider {
    * Restore offline operations into the queue
    */
   protected async restoreOfflineOperations(offlineLink: OfflineLink) {
+    // TODO This is a total mess - needs refactoring
     const offlineMutationHandler = new OfflineMutationsHandler(this.store,
       this.apolloClient as ApolloOfflineClient,
       this.config);
     offlineLink.setup(offlineMutationHandler);
     // Reschedule offline mutations for new client instance
     await offlineMutationHandler.replayOfflineMutations();
+
     // After pushing all online changes check and set network status
-    await offlineLink.initOnlineState();
+
+    // ^^ But why do we wait until now to check and set network status?
```

Comment:
Simply because if we let offline changes go without enqueueing then they will be:

- Non ordered (edit can be send faster than creation)
- Non retried (there is no protection for online changes now)
- Not going thru offline processor so no notifications etc.

We just wait to enqueue all offline changes and then initialize state. 
This means if we are offline - nothing happens if we are online offline scheduler will kick in and start replicating those. This is not perfect but it is the only way to avoid some more messy issues.